File: C++/Array/Kadane's Algorithm.cpp
class Solution{
    public:
    // arr: input array
    // n: size of array
    //Function to find the sum of contiguous subarray with maximum sum.
    int maxSubarraySum(int arr[], int n){
        
        // Your code here
        int maxSum = INT_MIN, sum = 0;
        for(int i = 0; i < n; i++){
            sum = sum + arr[i];
            maxSum = max(maxSum, sum);
            
            if(sum < 0){
                sum = 0;
            }
        }
        
        return maxSum;
        
    }
};

File: Python/Array/Kadane's Algorithm.py
class Solution:
    ##Complete this function
    #Function to find the sum of contiguous subarray with maximum sum.
    def maxSubArraySum(self,a,size):
        ans=0
        sum=0
        for i in range(len(a)):
            sum+=a[i]
            ans=max(ans,sum)
            if sum<0:
                sum=0
    
        return ans 
